systemtest/src/main/java/io/strimzi/systemtest/kafkaclients/KafkaClientProperties.java

```diff
@@ -0,0 +1,458 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.systemtest.kafkaclients;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import io.fabric8.kubernetes.api.model.Secret;
+import io.strimzi.kafka.oauth.common.HttpUtil;
+import io.strimzi.systemtest.Constants;
+import io.strimzi.systemtest.resources.ResourceManager;
+import io.strimzi.test.executor.Exec;
+import org.apache.kafka.clients.CommonClientConfigs;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.common.config.SaslConfigs;
+import org.apache.kafka.common.config.SslConfigs;
+import org.apache.kafka.common.security.auth.SecurityProtocol;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.security.InvalidParameterException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.util.Base64;
+import java.util.Iterator;
+import java.util.Properties;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static io.strimzi.kafka.oauth.common.OAuthAuthenticator.urlencode;
+import static io.strimzi.test.k8s.KubeClusterResource.kubeClient;
+import static org.junit.jupiter.api.Assertions.fail;
+
+/**
+ * Class KafkaClientProperties, which holds inner class builder for fluent way to invoke objects. It is used inside
+ * all our external clients such as BasicExternalKafkaClient or OauthExternalKafkaClient.
+ *
+ * Description of SuppressWarnings:
+ *
+ * caSecretName field is not initialized first inside KafkaClientProperties constructor and he is de-referenced
+ * in sharedClientProperties() method. This practically means, always make sure that before invoking this method
+ * you need first execute withCaSecretName().
```

Comment:
* There's no need to document these it in a javadoc comment, a normal comment will suffice. 
* The reasons can get stale. The interested reader can presumable try removing the suppressions and see what warnings they get and why.